server/routes/getSlashes.ts

```diff
@@ -0,0 +1,71 @@
+import Sequelize from 'sequelize';
+import { Request, Response, NextFunction } from 'express';
+import { factory, formatFilename } from '../../shared/logging';
+import { Errors } from './getOffences';
+
+const Op = Sequelize.Op;
+
+const getSlashes  = async (models, req: Request, res: Response, next: NextFunction) => {
+  const { chain, stash } = req.query;
+  let { startDate, endDate } = req.query;
+  let slashes;
+
+  if (!chain) return next(new Error(Errors.ChainIdNotFound));
+  const chainInfo = await models.Chain.findOne({
+    where: { id: chain }
+  });
+  if (!chainInfo) return next(new Error(Errors.InvalidChain));
+
+  // if date isn't defined we get for last 30 days
+  if (typeof startDate === 'undefined' || typeof endDate === 'undefined') {
+    const daysAgo = 30;
+    startDate = new Date(); // today
+    startDate.setDate(startDate.getDate() - daysAgo); // 30 days ago
+    startDate = new Date(startDate); // formatted
+    endDate = new Date(); // today
+  }
+  /**
+   * {
+        includeAddresses: [ 'Alice' ],
+        data: { kind: 'slash', validator: 'Alice', amount: '10000' },
+        blockNumber: 10
+    }
+   */
+  if (req.query.stash) {
```

Comment:
You're not reusing enough code imo. You should create the query string based on `if (req.query.stash)` but then reuse the same querying logic to execute that string. This way you don't need to have to same code repeated twice.

All your routes should adopt this paradigm.

Reply:
Thank you for letting me know this, I'll make sure I use this approach now on!